### PR TITLE
do not evict a daemonset pod when met eviction signal

### LINKED ISSUE #76930: do not evict a DaemonSet Pod
**What happened**:
a daemonset pod will be evicted by kubelet when met eviction signal. 
**What you expected to happen**:
do not evict a daemonset pod, since the Pod is immediately recreated and rescheduled back to the same node.
**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by Pingan2017**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
/kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #76930 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by Pingan2017** [comment]
why no cncf-cla label? :confused:  @lavalamp @smarterclayton

File: pkg/kubelet/eviction/eviction_manager.go [hunk 1]
```diff
@@ -563,6 +564,13 @@ func (m *managerImpl) evictPod(pod *v1.Pod, gracePeriodOverride int64, evictMsg
 			return false
 		}
 	}
+
+	if controllerRef := metav1.GetControllerOf(pod); controllerRef != nil {
+		if controllerRef.Kind == "DaemonSet" {
```
Comment (liggitt) [inline pkg/kubelet/eviction/eviction_manager.go]: we should not special-case daemonset pods at this level. nothing prevents someone from making their own node-targeting controller that functions identically to daemonsets

**Comment by liggitt** [comment]
/hold 

evicting daemonset pods can make sense when you are draining a node, and have added a taint that would prevent the daemonset from rescheduling a replacement pod on the node

**Comment by Pingan2017** [comment]
a daemonset pod is bound to a specific node. It can't reschedule to a new node even we evict  it.
https://kubernetes.io/docs/tasks/administer-cluster/out-of-resource/#daemonset

**Comment by liggitt** [comment]
> a daemonset pod is bound to a specific node. It can't reschedule to a new node even we evict it.

understood. eviction isn't always about rescheduling the pods, sometimes it's about draining a node. if the node is tainted, then daemonset pods on it are evicted, they will not be reassigned to the node by the scheduler because of the taint.

**Comment by liggitt** [comment]
/close

daemonset pods are not treated differently by virtue of being daemonset pods

**Pull request closed without merging**